server/main.py

```diff
@@ -0,0 +1,137 @@
+import os
+import string
+import random
+from fastapi import HTTPException
+app = FastAPI()
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=["*"],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+connection=create_connection(db_file)
+
+def generate_random_string(length=8):
+    letters = string.ascii_lowercase
+    return ''.join(random.choice(letters) for _ in range(length))
+
+@app.post("/upload/")
+async def update_file(key: Optional[str] = Form(None), encoded_content: List[str] = Form(...)):
+    try:
+        if not key:
+            key = generate_random_string()
+        if not encoded_content:
+            raise HTTPException(status_code=422, detail="Field 'encoded_content' cannot be empty")
+        key_directory = f"{key}"
+        directory_key = f"./storage/{key_directory}"
+        os.makedirs(directory_key , exist_ok=True)
+        for i, encoded_items in enumerate(encoded_content, start=1):
+                output_file_path = os.path.join(directory_key, f'encodedtxt{i}.txt')
+
+                with open(output_file_path, 'wb') as output_file:
+                    output_file.write(encoded_items.encode())
+        create_image_metadata( key, key_directory)
+        return JSONResponse(content={"message": "File uploaded successfully","key": key})
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.get("/get/{key}")
+async def retrieve_file(key: str, metadata_only: Optional[bool] = False):
+    try:
+        if not check_key_existence(key):
+            raise HTTPException(status_code=404, detail="Key not found")
+
+        filepath = get_metadata(key)
+        path = f"./storage/{filepath}"
+
+        if not path or not os.path.exists(path):
+           raise HTTPException(status_code=404, detail="Key not found")
+
+        binary_data_list = []
+
+        for filename in os.listdir(path):
+            file_path = os.path.join(path, filename)
+
+            if os.path.isfile(file_path):
+                with open(file_path, "rb") as file:
+                    binary_data = file.read()
+                    binary_data_list.append(binary_data)
+
+        return binary_data_list
+
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))   
+
+@app.put("/update/")
+async def update_files(key: str, encoded_content: List[str] = Form(...), new_key: Optional[str] = Form(None)):
+    filepath = get_metadata(key)
+
+    if filepath is None:
+        raise HTTPException(status_code=404, detail="Key not found")
+
+    path = f"./storage/{filepath}"
+
+    if new_key and new_key != key:
+ 
+        new_key_directory = f"./storage/{new_key}"
+        os.makedirs(new_key_directory, exist_ok=True)
+        create_image_metadata(new_key, new_key_directory)
+        path="./storage/"+get_metadata(key)
+        new_path = new_key_directory
+
+        for filename in os.listdir(path):
```

Comment:
I will create a tech debt ticket for this. We don't need to copy. Just updating the metadata is enough. To further elaborate, We will have to use the same directory as previous to store the new files and update only the meta data. Let the path remain same. Only Catch is that we need to suffix/prefix our folder names.

Reply:
No action required for this.